redash_emailer.py

```diff
@@ -62,31 +64,37 @@ def main(args):
     server.login(args.smtp_login, args.smtp_password)
 
     filename = 'query_%s_results.csv' % args.query_id
+    body_text = args.body
+    if len(rows) == 0:
+        body_text = body_text + '\n No data was returned; skipping attachment.'
 
     for recipient, rows in rows_by_recipient.items():
         msg = MIMEMultipart()
         msg['Subject'] = args.subject
         msg['From'] = args.from_address
         msg['To'] = recipient
-        msg.attach(MIMEText(args.body, 'plain'))
-        csv_file = io.StringIO()
-        csv_writer = csv.writer(csv_file, quoting=csv.QUOTE_NONNUMERIC)
-        keys = list(rows[0].keys())
-        csv_writer.writerow(keys)
-        for row in rows:
-            csv_writer.writerow([row[key] for key in keys])
-        csv_attachment = MIMEBase('application', 'octet-stream')
-        csv_attachment.set_payload(csv_file.getvalue())
-        encoders.encode_base64(csv_attachment)
-        csv_attachment.add_header('Content-Disposition', 'attachment',
-                                  filename=filename)
-        msg.attach(csv_attachment)
+        msg.attach(MIMEText(body_text, 'plain'))
+        if len(rows) > 0:
+            csv_file = io.StringIO()
+            csv_writer = csv.writer(csv_file, quoting=csv.QUOTE_NONNUMERIC)
```

Comment:
This looks fine, but just FYI, this could be a little cleaner with something like:
```
csv_writer = csv.DictWriter(csv_file, rows[0].keys(), quoting=csv.QUOTE_NONNUMERIC)
csv_writer.writeheader()
csv_writer.writerows(rows)
```